cameo/flux_analysis/analysis.py

```diff
@@ -367,6 +379,100 @@ class _PhenotypicPhasePlaneChunkEvaluator(object):
     def __init__(self, model, variable_reactions):
         self.model = model
         self.variable_reactions = variable_reactions
+        objective_reactions = [reaction for reaction in model.reactions
+                               if reaction.objective_coefficient != 0]
+        if len(objective_reactions) != 1:
+            raise Exception('multiple objectives not supported')
+        self.product_reaction = objective_reactions[0]
+        self.c_source_reactions = self.get_c_source_reactions()
+
+    def get_c_source_reactions(self):
+        """ carbon source reactions
+
+        Returns
+        -------
+        list
+           list of media reaction currently with negative flux or 0 if non defined
+        """
+        medium_reactions = [self.model.reactions.get_by_id(reaction) for reaction in self.model.medium.reaction_id]
+        self.model.solve()
+        source_reactions = [reaction for reaction in medium_reactions if
+                            (reaction.n_carbon > 0) and (reaction.flux < 0)]
```

Comment:
cheers @joaocardoso, learnt something, looks much better to me, however sadly yes it does get substantially slower that way..

using `solution.fluxes`
```{python}
In[112]: %time ppp2d = phenotypic_phase_plane(model, ['EX_o2_LPAREN_e_RPAREN_', 'EX_co2_LPAREN_e_RPAREN_'])

CPU times: user 17 s, sys: 80.3 ms, total: 17.1 s
Wall time: 17.2 s
In[113]: %time ppp = phenotypic_phase_plane(model, ['EX_o2_LPAREN_e_RPAREN_'])

CPU times: user 3.66 s, sys: 25 ms, total: 3.69 s
Wall time: 3.74 s
```

using `reaction.flux`
```{python}
In[114]: %time ppp = phenotypic_phase_plane(model, ['EX_o2_LPAREN_e_RPAREN_'])

CPU times: user 72.4 ms, sys: 1.73 ms, total: 74.2 ms
Wall time: 74.8 ms
In[115]: %time ppp2d = phenotypic_phase_plane(model, ['EX_o2_LPAREN_e_RPAREN_', 'EX_co2_LPAREN_e_RPAREN_'])

CPU times: user 867 ms, sys: 11 ms, total: 878 ms
Wall time: 881 ms
```